app/src/main/java/com/example/taskyy/domain/repository/AgendaRepository.kt

```diff
@@ -1,27 +1,47 @@
 package com.example.taskyy.domain.repository
 
-import android.content.Context
 import com.example.taskyy.domain.error.DataError
 import com.example.taskyy.domain.error.Result
 import com.example.taskyy.ui.objects.AgendaEventItem
 import com.example.taskyy.ui.objects.Reminder
+import com.example.taskyy.ui.objects.Task
 
 interface AgendaRepository {
     suspend fun logout(): Boolean
     suspend fun getUserName(email: String): String
+
+    //Reminder
     suspend fun saveReminderToDB(reminder: Reminder): Result<Reminder, DataError.Local>
     suspend fun saveReminderToApi(reminder: Reminder): Result<Reminder, DataError.Network>
     suspend fun updateReminderToApi(reminder: Reminder): Result<Reminder, DataError.Network>
     suspend fun getReminders(
         startDate: Long,
         endDate: Long
-    ): Result<List<Reminder>, DataError.Local>
+    ): Result<MutableList<Reminder>, DataError.Local>
```

Comment:
Same here, I don't yet see why this would be necessary. If you really need a mutable list somewhere, you can just convert a normal list to a mutable one at the place where you need it, but you don't need to change it all the way back to the origin of the list (your DB)